### PR TITLE
Deprecation cleanup

### LINKED ISSUE #259: Deprecation cleanup
Given the 2.0 bump, this is probably a good time to remove any deprecated methods. Also, ensure any reference documentation, AT's etc are updated.

### PR DESCRIPTION
**Pull request by chrisjs**
Resolves #259

### CODE REVIEW

File: src/test/java/org/springframework/cloud/deployer/spi/kubernetes/RunAbstractKubernetesDeployerTests.java [hunk 1]
```diff
@@ -152,65 +121,4 @@ public void requestMemory_deploymentProperty_usesDeploymentProperty() throws Exc
...
-
-	@Test
-	public void limitMemory_commonDeploymentProperty_mSuffix() throws Exception {
-		deploymentProperties.put(AppDeployer.MEMORY_PROPERTY_KEY, "512m");
-		Map<String, Quantity> limits = kubernetesDeployer.deduceResourceLimits(deploymentRequest);
-		MatcherAssert.assertThat(limits.get("memory"), is(new Quantity("512Mi")));
-	}
-
-	@Test
-	public void limitMemory_commonDeploymentProperty_gSuffix() throws Exception {
```
Comment (cppwfs) [inline src/test/java/org/springframework/cloud/deployer/spi/kubernetes/RunAbstractKubernetesDeployerTests.java]: Just curious why we removed gsuffix, wholeNumber and fraction tests

Comment (chrisjs) [reply src/test/java/org/springframework/cloud/deployer/spi/kubernetes/RunAbstractKubernetesDeployerTests.java]: the tests removed were ones using deprecated methods / properties. there are similar assertions that remained in the test class but use the non-deprecated equivalent methods / properties

File: src/main/java/org/springframework/cloud/deployer/spi/kubernetes/AbstractKubernetesDeployer.java [hunk 2]
```diff
@@ -255,67 +254,29 @@ protected PodSpec createPodSpec(String appId, AppDeploymentRequest request, Inte
...
-		}
+		String memory = request.getDeploymentProperties().get("spring.cloud.deployer.kubernetes.limits.memory");
 
-		// Use server property if there is no request setting
-		if (memOverride == null) {
-			if (properties.getLimits().getMemory() != null) {
-				memOverride = properties.getLimits().getMemory();
-			}
+		if (StringUtils.isEmpty(memory)) {
+			memory = properties.getLimits().getMemory();
```
[the remark this answers is not in the document]
Comment (chrisjs) [reply src/main/java/org/springframework/cloud/deployer/spi/kubernetes/AbstractKubernetesDeployer.java]: there isn't a default and fabric8 handles null fine which is the main consumer of these results

File: src/main/java/org/springframework/cloud/deployer/spi/kubernetes/AbstractKubernetesDeployer.java [hunk 3]
```diff
@@ -255,67 +254,29 @@ protected PodSpec createPodSpec(String appId, AppDeploymentRequest request, Inte
...
-		}
+		String cpu = request.getDeploymentProperties().get("spring.cloud.deployer.kubernetes.limits.cpu");
 
-		// Use server property if there is no request setting
-		if (cpuOverride == null) {
-			if (properties.getLimits().getCpu() != null) {
-				cpuOverride = properties.getLimits().getCpu();
-			}
+		if (StringUtils.isEmpty(cpu)) {
+			cpu = properties.getLimits().getCpu();
```
[the remark this answers is not in the document]
Comment (chrisjs) [reply src/main/java/org/springframework/cloud/deployer/spi/kubernetes/AbstractKubernetesDeployer.java]: same as above

**Comment by cppwfs** [comment]
LGTM

**Pull request merged**